target/linux/layerscape/patches-5.4/804-crypto-0016-MLKU-114-1-crypto-caam-reduce-page-0-regs-access-to-.patch

```diff
@@ -100,19 +100,20 @@ Signed-off-by: Horia Geantă <horia.geanta@nxp.com>
  		md_inst = mdha & CHA_VER_NUM_MASK;
 --- a/drivers/crypto/caam/caampkc.c
 +++ b/drivers/crypto/caam/caampkc.c
-@@ -1099,10 +1099,10 @@ int caam_pkc_init(struct device *ctrldev
+@@ -1099,11 +1099,11 @@ int caam_pkc_init(struct device *ctrldev
  
  	/* Determine public key hardware accelerator presence. */
- 	if (priv->era < 10)
+ 	if (priv->era < 10) {
 -		pk_inst = (rd_reg32(&priv->ctrl->perfmon.cha_num_ls) &
 +		pk_inst = (rd_reg32(&priv->jr[0]->perfmon.cha_num_ls) &
  			   CHA_ID_LS_PK_MASK) >> CHA_ID_LS_PK_SHIFT;
- 	else
--		pk_inst = rd_reg32(&priv->ctrl->vreg.pkha) & CHA_VER_NUM_MASK;
+ 	} else {
+ 		pkha = rd_reg32(&priv->ctrl->vreg.pkha);
+-		pk_inst = pkha & CHA_VER_NUM_MASK;
 +		pk_inst = rd_reg32(&priv->jr[0]->vreg.pkha) & CHA_VER_NUM_MASK;
```

Comment:
This change looks wrong to me, I think this part should look like this:
```
@@ -1099,10 +1099,10 @@ int caam_pkc_init(struct device *ctrldev
 
        /* Determine public key hardware accelerator presence. */
        if (priv->era < 10) {
-               pk_inst = (rd_reg32(&priv->ctrl->perfmon.cha_num_ls) &
+               pk_inst = (rd_reg32(&priv->jr[0]->perfmon.cha_num_ls) &
                           CHA_ID_LS_PK_MASK) >> CHA_ID_LS_PK_SHIFT;
        } else {
-               pkha = rd_reg32(&priv->ctrl->vreg.pkha);
+               pkha = rd_reg32(&priv->jr[0]->vreg.pkha);
                pk_inst = pkha & CHA_VER_NUM_MASK;
 
                /*
```

Reply:
Fixed and rebased.